src/NATS.Client.Core/Commands/ProtocolWriter.cs

```diff
@@ -46,159 +50,63 @@ public void WritePong()
     }
 
     // https://docs.nats.io/reference/reference-protocols/nats-protocol#pub
-    // PUB <subject> [reply-to] <#bytes>\r\n[payload]
-    // To omit the payload, set the payload size to 0, but the second CRLF is still required.
-    public void WritePublish(string subject, string? replyTo, ReadOnlySequence<byte> payload)
+    // PUB <subject> [reply-to] <#bytes>\r\n[payload]\r\n
+    public void WritePublish(string subject, string? replyTo, NatsHeaders? headers, ReadOnlySequence<byte> payload)
     {
-        var offset = 0;
```

Comment:
Removed precalculated `offset +=...` code since *BufferWriter.Advance()* method (which also marked with *AggressiveInlining*) does just that anyway and makes it much easier comprehend.